### ISSUE TITLE
AI cleanup order does not respect access restriction and specialized containers

### ISSUE THREAD

**Issue by Oukido** [Regalis11/Barotrauma#5812]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
AI with the cleanup command put items (for example ores that are stored in a metal crate) in the nearest available container, even if they have no matching id card for picked required. They also do ignore a present mineralcontainer that still has space but is further away.

**Steps To Reproduce**
Put a metalcrate filled with ore next to a secure steel cabinet with different picked required idcard than the bot possesses.
The bot will then put everything in the secure steel cabinet even without permission to access it.

**Version**
Windows v0.13.3.11

**Additional information**
I have noticed this happening while I was out mining and put the metalcrate in my airlock which is one level above my armory. Since I had one bot on cleanup duty it picked the items and put it in the access restricted secarmscab one level below (passing all the other available storage containers) since it was the nearest container in range of the metalcrate (ignoring walls and detours)

**Comment by itchyOwl** [comment]
The main issue of this ticket is fixed in Regalis11/Barotrauma-development@0c97c84.

I tested it by adding a secure locker that I set to be accessed only by the captain. To ensure that the bots try to prefer this container, I added the item identifier in the "Containable Restrictions" field. Then I started the game and spawned an item next to the container (doesn't matter which item as long as the identifier/tag matches what you filled in the "Containable Restrictions" field). Before the fix the bots would put the item in the secure container. After the fix, they'll find some other place for it.

![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/fe4ee6f2-5e34-4d3e-9c79-df261ab945dc)

![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/700f9862-a0a2-48a0-9323-66bb4eb44778)

The minor issue mentioned in the ticket about minerals not being placed in a container tagged as "mineralcontainer" is fixed in Regalis11/Barotrauma-development@866f956. The setup I used for testing it pictured below. Before the fix the bots would put the minerals in the closest container. After the fix they should prefer the locker tagged as "mineralcontainer".

![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/6621d6bf-ca51-41bf-b2fe-4fa031f0bbb5)

![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/2322c8d4-9d12-4d28-a480-a0ded67d9b10)
